community/modules/scripts/htcondor-install/files/autoscaler.py

```diff
@@ -272,47 +256,37 @@ def scale(self):
 
         if self.size < currentTarget:
             print("Scaling down. Looking for nodes that can be shut down")
-            # Find nodes that are not busy (all slots showing status as "Unclaimed")
-
-            idle_node_zones = {}
-            for slot_name in slot_names:
-                name_status = slot_name.split()
-                if len(name_status) > 1:
-                    name = name_status[0]
-                    status = name_status[1]
-                    zone = name_status[2]
-                    slot = "NO-SLOT"
-                    slot_server = name.split("@")
-                    if len(slot_server) > 1:
-                        slot = slot_server[0]
-                        server = slot_server[1].split(".")[0]
-                    else:
-                        server = slot_server[0].split(".")[0]
-
-                    if self.debug > 0:
-                        print(slot + ", " + server + ", " + status + "\n")
-
-                    if server not in idle_node_zones:
-                        if status == "Unclaimed":
-                            idle_node_zones[server] = zone
-                        else:
-                            idle_node_zones[server] = None
-                    else:
-                        if status != "Unclaimed":
-                            idle_node_zones[server] = None
+
+            # Find VMs that are unused (no dynamic slots created from
+            # partitionable slots) and have been booted for at least 150 seconds
+            max_daemon_start_time = int(datetime.datetime.now().timestamp()-150)
```

Comment:
I believe `time.time()` could replace `datetime.datetime.now().timestamp()` and would be more idiomatic python.

Reply:
I'm anticipating the need for `datetime` in future work, but I agree that `time.time()` is simpler.